### PR TITLE
Adding ToolTip with Animation v1

### PR DESCRIPTION
**Pull request by RedRosh**

### CODE REVIEW

**Review by Ignema — APPROVED** [review]

**Comment by Ignema** [comment]
LGTM

**Comment by ZynoZin** [comment]
Good work!

File: src/Roadmap/RoadmapNode.jsx [hunk 1]
```diff
@@ -1,14 +1,80 @@
...
 
-const RoadmapNode = () => {
-  return <img src={circle} alt="level" style={style}/>
-}
+const tooltipStyle = {
+  backgroundColor: "black",
+  color: "white",
+  opacity: 0.8,
+  zIndex: 100,
+  width: "300px",
```
Comment (assalielmehdi) [inline src/Roadmap/RoadmapNode.jsx:17]: px is the defaut when you use nubmers.

```
width: 300,
```

File: src/Roadmap/RoadmapNode.jsx [hunk 2]
```diff
@@ -1,14 +1,80 @@
...
+  padding: "1rem",
```
Comment (assalielmehdi) [inline src/Roadmap/RoadmapNode.jsx:18]: rem is good, but we have to stay consistent, either we use px or rem for all metrics

File: src/Roadmap/RoadmapNode.jsx [hunk 3]
```diff
@@ -1,14 +1,80 @@
...
+  borderRadius: 10,
+};
+
+const RoadmapNode = ({ gridNode }) => {
+  //* Animation config
+  const config = { tension: 300, friction: 15 };
+  const initialStyles = { opacity: 0 };
+  const [props, setSpring] = useSpring(() => initialStyles);
+
+  function onMount() {
```
Comment (assalielmehdi) [inline src/Roadmap/RoadmapNode.jsx:29]: Refactor to fat arrow function, to be consistent w/ the rest of the code base

File: src/Roadmap/RoadmapNode.jsx [hunk 4]
```diff
@@ -1,14 +1,80 @@
...
+    setSpring({
+      opacity: 1,
+      onRest: () => {},
+      config,
+    });
+  }
+
+  function onHide({ unmount }) {
```
Comment (assalielmehdi) [inline src/Roadmap/RoadmapNode.jsx:37]: Same here

File: src/Roadmap/RoadmapNode.jsx [hunk 5]
```diff
@@ -1,14 +1,80 @@
...
+    setSpring({
+      ...initialStyles,
+      onRest: unmount,
+      config: { ...config, clamp: true },
+    });
+  }
+
+  return (
+    <Tippy
+      interactive={true}
```
Comment (assalielmehdi) [inline src/Roadmap/RoadmapNode.jsx:47]: No need to specify `={true}` for boolean props

```
<Tippy interactive />
```

**Pull request merged**